charmcraft/commands/build.py

```diff
@@ -174,23 +174,22 @@ def build_charm(self, bases_config: BasesConfiguration) -> str:
         logger.info("Created '%s'.", zipname)
         return zipname
 
-    def run(self, bases_indices: Optional[List[int]] = None) -> List[str]:
+    def run(
+        self, bases_indices: Optional[List[int]] = None, destructive_mode: bool = False
+    ) -> List[str]:
         """Run build process.
 
-        In managed-mode (and eventually destructive-mode), build for each bases
-        configuration which has a matching build-on to the host we are executing
-        on.  Warn for each base configuration that is incompatible.  Error if
-        unable to produce any builds for any bases configuration.
-
-        Otherwise, build the charm in the legacy mode until we have complete
-        provider support and can drop it.
+        In managed-mode or destructive-mode, build for each bases configuration
+        which has a matching build-on to the host we are executing on.  Warn for
+        each base configuration that is incompatible.  Error if unable to
+        produce any builds for any bases configuration.
 
         :returns: List of charm files created.
         """
         charms: List[str] = []
 
         is_managed_mode = is_charmcraft_running_in_managed_mode()
-        if not is_managed_mode:
+        if not is_managed_mode and not destructive_mode:
```

Comment:
`is_managed_mode` and `destructive_mode` look inconsistent, maybe use `managed_mode` instead?

Reply:
Agreed, noted this as well to follow up with but did not want to add the refactoring in this PR.

Reply:
But I will in a separate commit ;)

Reply:
Done!